src/publisher/sqs.ts

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+import AWS = require('aws-sdk');
+const { Consumer } = require('sqs-consumer');
+import { SQS } from 'aws-sdk';
+import { AWSConfig } from '../aws/config';
+import { Message } from '../model/message';
+import { HasSubscribers, Publisher, Subscriber, Subscription } from './publisher';
+
+/**
+ * A Publisher that use SQS to message communications.
+ */
+export class SQSPublisher implements Publisher, HasSubscribers {
+    private url: string;
+    private sqs: SQS;
+
+    constructor(url: string, awsconfig: AWSConfig) {
+        AWS.config.update(awsconfig);
+        this.sqs = new AWS.SQS();
+        this.url = url;
+    }
+
+    public async publish(message: Message): Promise<boolean> {
+        const sqsData = {
+            MessageAttributes: {
+                "aggregation": {
+                    DataType: "String",
+                    StringValue: message.stream.aggregation
+                },
+                "commitTimestamp": {
+                    DataType: "Number",
+                    StringValue: `${message.event.commitTimestamp}`
+                },
+                "id": {
+                    DataType: "String",
+                    StringValue: message.stream.id,
+                },
+            },
+            MessageBody: JSON.stringify(message),
+            MessageDeduplicationId: `${message.stream.aggregation}:${message.stream.id}:${message.event.commitTimestamp}`,
+            MessageGroupId: message.stream.aggregation,
```

Comment:
groupid should only be used with fifo queues. If you want to implement a pub/sub notification for multiples distinct groups of subscribers, you can not specify a group id. I think you shoud allow the eventstore user to tell which kind of queue to use (standard or fifo) and set the configuration property accordingly